draft-birkholz-scitt-architecture.md

```diff
@@ -552,64 +575,126 @@ Editor's Note: This may be moved to appendix.
+### Retrieve Operation Status
 
-[TODO] probably a bad idea to define a new header, or is it ok? can we register a new one? https://www.iana.org/assignments/http-fields/http-fields.xhtml
+#### Request
 
-The `400` response has a `Content-Type: application/json` header and a body containing details about the error:
+~~~
+GET <Base URL>/operations/<Operation ID>
+~~~
 
-```json
-{
-  "error": {
-    "code": "<error code>",
-    "message": "<message>"
-  }
-}
-```
+#### Response
 
-`AwaitingDIDResolutionTryLater` means the service does not have an up-to-date DID document of the DID referenced in the Signed Claims but is performing or will perform a DID resolution after which the client may retry the request. The response may contain the HTTP header `Retry-After` to inform the client about the expected wait time.
+One of the following:
 
-`InvalidInput` means either the Signed Claims message is syntactically malformed, violates the signing profile (e.g. signing algorithm), or has an invalid signature relative to the currently resolved DID document.
+- Status 200 - Registration is pending
+    - Header: `Content-Type: application/json`
+    - (Optional) Header: `Retry-After: <seconds>`
+    - Body: `{ "operationId": "<Operation ID>", "status": "pending" }`
 
-### Retrieve Registration Receipt
+- Status 200 - Registration was successful
+    - Header: `Location: <Base URL>/entries/<Entry ID>`
+    - Header: `Content-Type: application/json`
+    - Body: `{ "operationId": "<Operation ID>", "status": "registered", "entryId": "<Entry ID>" }`
+
+- Status 200 - Registration failed
+    - Header `Content-Type: application/json`
+    - Body: `{ "operationId": "<Operation ID>", "status": "failed", "error": { "code": "<code>", "message": "<message>" } }`
+    - Error code: `InvalidInput`
+
+- Status 404 - Unknown Operation ID
+    - Error code: `NotFound`
+    - This can happen if the operation ID has expired and been deleted.
+
+If an operation ID is invalid (i.e., it does not correspond to any submit operation), a service may return either a 404 or a `pending` status. This is because differentiating between the two may not be possible in an eventually consistent system.
+
+### Retrieve Registration Entry
 
 #### Request
 
 ~~~
-GET <Base URL>/entries/<Transaction ID>/receipt
+GET <Base URL>/entries/<Entry ID>
 ~~~
 
 #### Response
 
 One of the following:
 
-- HTTP Status `200` - Registration was successful and the Receipt is returned.
-- HTTP Status `400` - Transaction exists but does not correspond to a Registration Request.
-  - Error code `TransactionMismatch`
-- HTTP Status `404` - Transaction is pending, unknown, or invalid.
-  - Error code `TransactionPendingOrUnknown`
-  - Error code `TransactionInvalid`
+- Status 200.
+  - Header `Content-Type: application/json`
+  - Body `{ "entryId": "<Entry ID>", "status": "registered" }`
```

Comment:
Probably `"status"` should be removed from this response since any other case would return a 404 anyway.

Reply:
NIST labeling requires a date. I suggest replacing status with EffectiveDate.

Reply:
I removed `"status"` without adding more fields for now. Registration date is part of the receipt already, but we can discuss in separate issues what kind of metadata this entry resource should have.